packages/@storefront/flux-capacitor/src/core/store/middleware.ts

```diff
@@ -138,13 +153,13 @@ export namespace Middleware {
         batchLevel++;
       } else if (action.type === POP) {
         batchLevel--;
-      } else if (SAVE_STATE_ACTIONS.includes(action.type) && !hasDispatched) {
+      } else if (SAVE_STATE_ACTIONS.includes(action.type)) {
         hasDispatched = true;
-        dispatch({ type: Actions.SAVE_STATE });
       }
 
-      if (batchLevel === 0) {
+      if (batchLevel === 0 && hasDispatched) {
         hasDispatched = false;
+        dispatch({ type: Actions.SAVE_STATE });
```

Comment:
Flagging for later discussion. I'm not sure if this change is necessary or correct.

Reply:
I'm p sure it's not needed. Will take it out.